Engineering/Frontend/CSS in JS.md

```diff
@@ -0,0 +1,41 @@
+---
+tags: frontend, css-in-js
+author: Tran khac Vy
+---
+
+**CSS-in-JS** is a styling technique where Javascript is used to style the component. These are multiple implementation of this concept in the form of libraries such as [emotion](https://emotion.sh), [styled-component](https://styled-components.com/), [JSS](https://cssinjs.org). They airm to tacke the limitations of CSS, such as lack of dynamic functionality, scoping, and portability.
+
+## Benefits of using CSS-in-JS
+
+### Scoping
+- With CSS-in-JS, writing new styles cannot affect anything else in other places on the site, so there’s no need to worry about writing a style that has bad or unintended consequences elsewhere due to a selector in the global scope.
+
+### Avoid naming collisions
+- CSS-in-JS libraries automaticly generate unique selectors for what’s being styled. We don't need to think about name.
+
+### Dynamic Functionality
+- As CSS-in-JS is essentially JavaScript code, you can apply complex logic to your style rules, such as loops, conditionals, variables, state-based styling, and more.
+
+### Dead code elimination
+- CSS-in-JS helps with removing dead code. The only styles that are loaded are the styles for the components in use at any given time. There’s no shipping any unused styles. When a component dies, so does its styles.
+
+### Developer ergonomics
+- It can be nice to have styles in the same file (or otherwise very close to) the component itself. In the same way, some developers feel very comfortable in JSX. Also being able to style things without any scoping worry means developers may feel empowered about styling rather than intimidated by it.
+
+## Disadvantages of CSS-in-JS
```

Comment:
I think adding an extra bundle size might be worth mentioning?

Reply:
Added. `Adding library => more code => increase bundle size.`
Do you have any suggestions for a more meaningful full sentence?